lib/supportsFeature.js

```diff
@@ -1,31 +1,63 @@
 const features = require('../minecraft-data/data/pc/common/features.json')
-
-const versions = {
-  pc: require('../minecraft-data/data/pc/common/protocolVersions.json'),
-  bedrock: require('../minecraft-data/data/bedrock/common/protocolVersions.json')
-}
-const versionList = versions.pc
-const versionToIndex = Object.fromEntries(versionList.map((version, i) => [version.minecraftVersion, versionList.length - i]))
 const nameToFeature = Object.fromEntries(features.map(feature => [feature.name, feature]))
 
-module.exports = (featureName, minecraftVersion) => {
+function isFeatureInRange (featureName, versionObj) {
   const feature = nameToFeature[featureName]
   if (feature === undefined) {
     throw new Error(`Feature ${feature} doesn't exist`)
   }
 
-  const currentVer = versionToIndex[minecraftVersion]
-  const minVer = versionToIndex[feature.versions[0]]
-  const maxVer = versionToIndex[feature.versions[1]]
-  if (currentVer === undefined) {
-    throw new Error(`Version ${minecraftVersion} doesn't exist`)
+  if (feature.values) {
+    for (const { value, versions, version } of feature.values) { // we're using feature.version
+      if (version) {
+        const ver = version.replace('_major', '')
+        if (!/^\d\.\d+$/.test(ver)) {
+          throw new Error(`Not a correct major version value, instead the version is: "${version}"`)
+        }
+        if (versionObj.majorVersion === ver) {
+          return value
+        }
+      } else { // we're using feature.versions
+        const [minVer, maxVer] = versions
+        if (isVersionInRange(minVer, versionObj, maxVer)) {
+          return value
+        }
+      }
+    }
+    return null // if we didn't match anything, return null
+  } else {
+    const [minVer, maxVer] = feature.versions
+    return isVersionInRange(minVer, versionObj, maxVer)
+  }
+}
+
+function isVersionInRange (minVer, versionObj, maxVer) {
+  let inRange = true
+  const { majorVersion } = versionObj
+  if (minVer.endsWith('_major')) {
+    const ver = removeMajorSuffix(minVer)
+    inRange = inRange && getVersionObj(majorVersion)['>='](ver)
```

Comment:
getVersionObj(majorVersion) this is either the smallest, the largest or something in middle of the range
in all three cases, it will be incorrect for use in here or in line 48